lib/Conversion/TorchToTcp/TcpCustomOp.cpp

```diff
@@ -117,6 +117,147 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
+            op, "Zero point operation is not ValueTensorLiteralOp or Zero "
+                "operation");
+      auto zeroPointElements =
+          dyn_cast<DenseIntElementsAttr>(zeroPointTensor.getValueAttr());
+      if (!zeroPointElements || zeroPointElements.getNumElements() != 1)
+        return rewriter.notifyMatchFailure(
+            op, "Unsupported zero point type or size");
+      zeroPoint = (*zeroPointElements.begin()).getSExtValue();
+    }
+    helper.addDenseIntArrayAttr("zero_point", {zeroPoint});
+
+    return helper.replace();
+  }
+};
+
+class ConvertAtenFakeQuantizePerChannelAffineOp
+    : public OpConversionPattern<AtenFakeQuantizePerChannelAffineOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenFakeQuantizePerChannelAffineOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    torch_to_tcp::TorchToTcpCustomOpConversionHelper helper{op, rewriter,
+                                                            getTypeConverter()};
+    helper.addOperand("self", adaptor.getSelf());
+    helper.addIntAttr("axis", op.getQuantMin());
+    helper.addIntAttr("quant_min", op.getQuantMin());
+    helper.addIntAttr("quant_max", op.getQuantMax());
+
+    // scale
+    auto scaleOp = op.getScale().getDefiningOp();
+    if (!scaleOp)
+      return rewriter.notifyMatchFailure(op, "Missing scale operation");
+    auto scaleTensor = dyn_cast<torch::Torch::ValueTensorLiteralOp>(scaleOp);
+    if (!scaleTensor)
+      return rewriter.notifyMatchFailure(
+          op, "Scale operation is not ValueTensorLiteralOp");
+    auto scaleElements =
+        dyn_cast<DenseFPElementsAttr>(scaleTensor.getValueAttr());
+    if (!scaleElements || scaleElements.getType().getShape().size() != 1)
+      return rewriter.notifyMatchFailure(op, "Unsupported scale type or size");
+    SmallVector<double> scale;
+    for (auto val : scaleElements.getValues<APFloat>())
+      scale.push_back(val.convertToDouble());
+    helper.addDenseFloatArrayAttr("scale", scale);
+
+    // zero_point
+    auto zeroPointOp = op.getZeroPoint().getDefiningOp();
+    SmallVector<int64_t> zeroPoint;
+    if (!zeroPointOp)
+      return rewriter.notifyMatchFailure(op, "Missing zero point operation");
+    if (dyn_cast<torch::Torch::AtenZerosOp>(zeroPointOp) ||
+        dyn_cast<torch::Torch::AtenZerosLikeOp>(zeroPointOp)) {
+      zeroPoint.assign(scale.size(), 0);
+    } else {
+      auto zeroPointTensor =
+          dyn_cast<torch::Torch::ValueTensorLiteralOp>(zeroPointOp);
+      if (!zeroPointTensor)
+        return rewriter.notifyMatchFailure(
+            op, "Zero point operation is not ValueTensorLiteralOp or Zero "
+                "operation");
+      auto zeroPointElements =
+          dyn_cast<DenseIntElementsAttr>(zeroPointTensor.getValueAttr());
+      if (!zeroPointElements ||
+          zeroPointElements.getType().getShape().size() != 1)
```

Comment:
Curious, is this checking that `zeroPointElements` is a 1D tensor? Please add a comment for clarity.